### PR TITLE
fix: make request event compatible with request

### LINKED ISSUE #14: compat: Expects 'response' event to be httpIncomingMessage, got a PassThrough stream
When making a streaming request, `teeny-request` returns a `PassThrough` stream that emits the `response` event that contains `res.body` from `node-fetch`.

This is incompatible with `request`, since the event is expected to contain `http.IncomingMessage`:

https://github.com/request/request#streaming
https://nodejs.org/api/http.html#http_event_response

### PR DESCRIPTION
**Pull request by kinwa91**
fixed #14

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -234,9 +234,15 @@ function teenyRequest(reqOpts: r.Options, callback?: r.RequestCallback) {
...
 teenyRequest.defaults = (defaults: r.Options) => {
-  return (reqOpts: r.Options, callback: r.RequestCallback) => {
-    return teenyRequest({...defaults, ...reqOpts}, callback);
-  };
+  return (reqOpts: r.Options, callback?: r.RequestCallback): PassThrough|
+      void => {
+        const opts = {...defaults, ...reqOpts};
+        if (callback === undefined) {
+          return teenyRequest(opts);
+        } else {
```
Comment (JustinBeckwith) [inline src/index.ts]: You can drop the else here since it returns above :)

File: src/index.ts [hunk 2]
```diff
@@ -168,20 +170,18 @@ function teenyRequest(reqOpts: r.Options, callback?: r.RequestCallback) {
             return;
           }
 
-          const encoding = res.headers.get('content-encoding');
```
Comment (JustinBeckwith) [inline src/index.ts]: Can you explain why this is getting deleted? I just don't understand what it was doing is all :)

File: test/index.ts [hunk 3]
```diff
@@ -1,7 +1,13 @@
 import * as assert from 'assert';
+import * as nock from 'nock';
 import * as r from 'request';
+
 import {teenyRequest} from '../src';
 
+nock('http://www.example.com').persist().get('/').reply(202, 'ok', {
```
Comment (JustinBeckwith) [inline test/index.ts]: Persist is bad. Instead, please create a function that returns the scope, then check `scope.done()` in the function.  This helps a.) ensure the requested nock was actually executed and b.) ensure it was only executed the # of times you expect it to be :)

File: src/index.ts (hunk 2 above)
Comment (kinwa91) [reply src/index.ts]: Previously, the `response` event emitted `res.body`, and this adds a "headers" property to that object that only contains the `content-encoding` header.

As we found out, `res.body` is not the correct event message for `response`, so we emit an object that is similar to `http.IncomingMessage`, which is what the `request` module emits, with the properties `headers`, `statusMessage` and `statusCode`.

So I just take the response header from `node-fetch` and emit back out.

File: src/index.ts (hunk 1 above)
Comment (kinwa91) [reply src/index.ts]: fixed :)

File: test/index.ts (hunk 3 above)
Comment (kinwa91) [reply test/index.ts]: fixed :)

**Pull request merged**